pkg/client/client.go

```diff
@@ -383,6 +384,31 @@ func (c *Client) Fetch(ctx context.Context, request FetchRequest) (res *FetchRes
 	ctx, spanEnder := telemetry.StartSpanFromContext(ctx, "Fetch")
 	defer spanEnder(retErr)
 
+	{
```

Comment:
would move this to a function, but also just to request.Providers a function that returns providers/count

Reply:
Fixed